templates/common/vsphere/files/vsphere-hostname.yaml

```diff
@@ -0,0 +1,12 @@
+filesystem: "root"
+mode: 0755
+path: "/usr/local/bin/vsphere-hostname.sh"
+contents:
+  inline: |
+    #!/usr/bin/env bash
+    set -e
+
+    if hostname=$(/bin/vmtoolsd --cmd 'info-get guestinfo.hostname'); then
+        /usr/bin/hostnamectl --transient --static set-hostname ${hostname}
```

Comment:
```
 hostnamectl set-hostname --help
hostnamectl [OPTIONS...] COMMAND ...

Query or change system hostname.

  -h --help              Show this help
     --version           Show package version
     --no-ask-password   Do not prompt for password
  -H --host=[USER@]HOST  Operate on remote host
  -M --machine=CONTAINER Operate on local container
     --transient         Only set transient hostname
     --static            Only set static hostname
     --pretty            Only set pretty hostname
```

is providing both `transient` and `static` valid option?

Reply:
yep...tested and reviewed man page.